java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -88,13 +97,68 @@ public JsonFileReader(File inputFile, BufferAllocator allocator) throws JsonPars
     this.parser = jsonFactory.createParser(inputFile);
   }
 
+  @Override
+  public Dictionary lookup(long id) {
+    if (initialized) {
+      return dictionaries.get(id);
+    } else {
+      return null;
+    }
+  }
+
   public Schema start() throws JsonParseException, IOException {
     readToken(START_OBJECT);
     {
-      this.schema = readNextField("schema", Schema.class);
+      Schema originalSchema = readNextField("schema", Schema.class);
+      List<Field> fields = new ArrayList<>();
+      dictionaries = new HashMap<>();
+
+      // Convert fields with dictionaries to have the index type
+      for (Field field: originalSchema.getFields()) {
```

Comment:
Is there a JIRA about handling encoded subfields? We should tackle this sooner rather than later. As long as this doesn't leak into the user API I think it's OK